### PR TITLE
[DEVX:122] Added Workflow Interface

### PR DESCRIPTION
**Pull request by sainivedh**
## What
 - Added functionality to interface with Workflows(predict, list, versions)
 
 ## How
### Workflow Predict
```python
# Note: CLARIFAI_PAT must be set as env variable.
from clarifai.client.workflow import Workflow

# Workflow Predict
workflow = Workflow(user_id="user_id", app_id="app_id", workflow_id="model_id")
workflow_prediction = workflow.predict_by_url(url="url", input_type="image") # Supports image, text, audio, video

# Customizing Workflow Inference Output
workflow = Workflow(user_id="user_id", app_id="app_id", workflow_id="workflow_id",
                  output_config={"min_value": 0.98}) # Return predictions having prediction confidence > 0.98
workflow_prediction = workflow.predict_by_filepath(filepath="local_filepath", input_type="text") # Supports image, text, audio, video
```

### Workflows Listing
```python
# Note: CLARIFAI_PAT must be set as env variable.

# List all workflow versions
all_workflow_versions = workflow.list_versions()

# Go to specific workflow version
workflow_v1 = Workflow(workflow_id="workflow_id", workflow_version=dict(id="workflow_version_id"), app_id="app_id", user_id="user_id")

# List all workflow in an app
all_workflow = app.list_workflow()

# List all workflow in community filtered by description
all_face_community_workflows = App().list_workflow(filter_by={"query": "face"}, only_in_app=False) # Get all face related workflows
```

## Tests

[Unit tests](https://github.com/Clarifai/clarifai-python/blob/604c9b7ca1110ad3daa42d613d9583334fe0caa0/tests/test_workflow_predict.py) added for Workflow predict with different scenarios
 - Predict by url, bytes
 - Predict by various output configs (max_concepts)
 
 ## Note
 Skipping Workflow unit tests in build as secrets needed for PAT are not yet setup for this repo

### CODE REVIEW

File: clarifai/client/workflow.py [hunk 1]
```diff
@@ -1,22 +1,167 @@
...
   Inherits from BaseClient for authentication purposes.
   """
 
-  def __init__(self, workflow_id: str, **kwargs):
+  def __init__(self,
+               workflow_id: str = "",
+               workflow_version: Dict = {'id': ""},
+               output_config: Dict = {'min_value': 0},
+               **kwargs):
     """Initializes an Workflow object.
```
Comment (ackizilkale) [inline clarifai/client/workflow.py:25]: a Workflow

File: README.md [hunk 2]
```diff
@@ -72,3 +73,37 @@ all_models = app.list_models()
...
+# Note: CLARIFAI_PAT must be set as env variable.
+from clarifai.client.workflow import Workflow
+
+# Workflow Predict
+workflow = Workflow(user_id="user_id", app_id="app_id", workflow_id="model_id")
+workflow_prediction = workflow.predict_by_url(url="url", input_type="image") # Supports image, text, audio, video
+
+# Customizing Workflow Inference Output
+workflow = Workflow(user_id="user_id", app_id="app_id", workflow_id="workflow_id",
+                  output_config={"min_value": 0.98}) # Return predictions having prediction confidence > 0.98
```
Comment (ackizilkale) [inline README.md:90]: Question out of curiosity. How does this min_value work if I have multiple models in a workflow? Do all the models in my workflow need to satisfy this constraint I define here?

**Review by ackizilkale — APPROVED** [review]
looks good. a very minor spelling issue above.

**Review by stmugisha — APPROVED** [review]
LGTM

File: README.md [hunk 3]
```diff
@@ -72,3 +73,37 @@ all_models = app.list_models()
...
+
+## Interacting with Workflows
+
+### Workflow Predict
+```python
+# Note: CLARIFAI_PAT must be set as env variable.
+from clarifai.client.workflow import Workflow
+
+# Workflow Predict
+workflow = Workflow(user_id="user_id", app_id="app_id", workflow_id="model_id")
```
Comment (sanjaychelliah) [inline README.md:85]: Isn't it `workflow_id`??

File: clarifai/client/workflow.py [hunk 4]
```diff
@@ -1,22 +1,167 @@
...
+      raise UserError('Invalid input type it should be image, text, video or audio.')
+    if not isinstance(input_bytes, bytes):
+      raise UserError('Invalid bytes.')
+
+    if input_type == "image":
+      input_proto = resources_pb2.Input(
+          data=resources_pb2.Data(image=resources_pb2.Image(base64=input_bytes)))
+    elif input_type == "text":
+      input_proto = resources_pb2.Input(
+          data=resources_pb2.Data(text=resources_pb2.Text(raw=input_bytes)))
```
Comment (sanjaychelliah) [inline clarifai/client/workflow.py:100]: Just checking, is this working for text?

**Review by sanjaychelliah — COMMENTED** [review]
LGTM! minor comments

File: README.md (hunk 2 above)
Comment (sainivedh) [reply README.md:90]: According to [endpoint](https://github.com/Clarifai/clarifai/blob/4ec4141c6c5cecd371ea77672e2a6bcce50f4f31/proto/clarifai/api/service.proto#L9152) comments I guess this output_config applies to all models in workflow.

**Pull request merged**